### ISSUE TITLE
[Android] AutomationId still does not work

### ISSUE THREAD

**Issue by GeorgeDarakchiev** [dotnet/maui#8277]
### Description

The issues [#4714 ](https://github.com/dotnet/maui/issues/4714) and [#3260](https://github.com/dotnet/maui/issues/3260) are marked as resolved, but they can still be reproduced with the latest version 6.0.400 (Service Release 1).

Please note, that in order for this to be fixed the ContentDescription needs to be set (setting ViewIdResourceName will not help). The reason behind this is that ultimately the Android[ UIAutomator tool](https://developer.android.com/reference/androidx/test/uiautomator/UiSelector) is looking for  content-description in the Android application.

This was working correctly with Preview 11 and haven't worked since then.

### Steps to Reproduce

[TestAutomationId.zip](https://github.com/dotnet/maui/files/8966961/TestAutomationId.zip)

You can test it with the attached project, created with new MauiApp + added some elements and AutomationIds.

1. Run the the app on Android
2. Open Appium and check the content-description of the elements

For more details on how to use Appium to check for content-description see [here ](https://github.com/dotnet/maui/issues/4714#issuecomment-1112928796)

### Version with bug

6.0.400 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

All

### Did you find any workaround?

As a workaround SemanticProperties.Description can be set, but this leads to issues on iOS.

**Comment by GeorgeDarakchiev** [comment]
@PureWeen and @rmarinho if you prefer for us to move this conversation in the old item #4714 , please reopen it so that I can write there. I'm currently unable to do so, because it is marked as resolved.

**Comment by PureWeen** [comment]
@GeorgeDarakchiev can you please provide a full sample with the appium parts as well? 

When I tested this change using `FindElementByAccessibilityId` if worked fine

If you're wanting to check the contentDesc directly instead of using `FindElementByAccessibilityId ` I don't think we'll be able to make that work. 

Unless I'm mistaken and `FindElementByAccessibilityId` isn't working for you? In which case can you please attach a full repro with the appium bits?

**Comment by GeorgeDarakchiev** [comment]
Hi @PureWeen 

Here are some steps that you can use including the `FindElementByAccessibilityId `:

1. Run https://github.com/dotnet/maui/files/8966961/TestAutomationId.zip project on Android
2. Start Appium and connect to the Android emulator
3. Tap on the Search icon
4. Choose the locator to be Accessibility Id and search with any AutomationId set in the app, for example "HelloLabel"
-> No Elements are found.

Here is a video: 

https://user-images.githubusercontent.com/28561290/175474173-6783522c-ff8f-4326-9004-54977f02ba39.mp4


I am also seeing this in my tests where I am using the `FindElementByAccessibilityId `.

Please note the following two things:

1. I don't see how `FindElementByAccessibilityId ` can work without content-description. Check the [Appium documentations](https://appium.io/docs/en/commands/element/find-elements/): 
`Accessibility ID - Read a unique identifier for a UI element. For XCUITest it is the element's accessibility-id attribute. For Android it is the element's content-desc attribute.`
1. If you have tested using the https://github.com/dotnet/maui/files/8586594/MauiApp27.zip app from the old issues, please look at the "App.xaml.cs" where the ContentDescription is set: 
`androidView.ContentDescription = view.AutomationId;`

I hope this helps.

**Comment by PureWeen** [comment]
@GeorgeDarakchiev can you use `findElementById` instead of `FindElementByAccessibilityId` ?

**Comment by GeorgeDarakchiev** [comment]
Hi @PureWeen,

Using `findElementById` is not feasible due to the following reasons:

1. The id of the element is not the AutomationId set in the app. It looks like the name of the package is added, for example: `com.companyname.testautomationid:id/image`. Also as we are controls vendor we would like to have the opportunity to set AutomationId to part of the controls so they can be accessible for our clients and be the same in all apps used. 
For example the RadComboBox control has a DropDownButton that we can set AutomationId to. Then I would expect the user to be able to use this AutomationId in all applications RadComboBox is added to and having the opportunity to create a method `TapDropDownButton() `that can be reused across different applications tests. Currently this cannot be achieved.

1. It doesn't seem right to me that I should use findElementById on Android and use FindElementByAccessibilityId for all other platforms. After all .Net Maui is a cross-platform solution and setting AutomationId should be consistent across platforms.

**Comment by 4sunshine4** [comment]
Any news on this issue? We're facing the same problem, switching from Xamarin to Maui. We can't access Elements by FindElementByAccessibilityId(accessibilityId) anymore. As mentioned above, seems we have to add the package, at least if we want to search by ID (app,package:id/accessibilityId) - the automationid property itself seems to be lost?

**Comment by sroesel** [comment]
Hi, just created another example-repo: https://github.com/sroesel/maui-automationids-not-working
You can see the behaviour of maui as well as xamarin with AutomationIds. Screenshots in root-folder. Created with latest VS-Preview! 

Any progress on this issue would be great, thx :)

**Comment by PureWeen** [comment]
We most likely aren't going to change how this works unless someone find a solution I'm not aware of. With XF `AutomationId` was mapped to `contentDesc` which wreaked havoc on Accessibility. 

We have a set of extensions that we're hoping to make a bit more public for `Appium` use as we've started using `appium` internally
https://github.com/dotnet/maui/blob/main/src/TestUtils/src/TestUtils.Appium.UITests/AppiumUITestApp.cs

If you want to map automationid to `ContentDescription`

Then I'd recommend tapping into the `ViewHandler.Mapper` and routing `AutomationId` yourself to contentDesc on `Android`

**Comment by homeyf** [comment]
Need install Appium Server: the third-party drivers.

**Comment by jfversluis** [comment]
> We have a set of extensions that we're hoping to make a bit more public for Appium use as we've started using appium internally
https://github.com/dotnet/maui/blob/main/src/TestUtils/src/TestUtils.Appium.UITests/AppiumUITestApp.cs

Those can now be found under https://github.com/jfversluis/Plugin.Maui.UITestHelpers

For the rest I think we have made it clear here that unless someone comes with a very good reason we're not going to change this behavior. Since that hasn't happened yet I will be closing this for now.
